src/common/transformations/src/transformations/common_optimizations/optimize_gather_nd.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/optimize_gather_nd.hpp"
+
+#include <memory>
+#include <ngraph/op/util/op_types.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/op/util/gather_nd_base.hpp>
+
+#include "itt.hpp"
+
+ov::pass::OptimizerGatherND::OptimizerGatherND() {
+    MATCHER_SCOPE(OptimizerGatherND);
+    auto gather_nd =
+        pattern::wrap_type<ov::op::util::GatherNDBase>({pattern::any_input(), pattern::wrap_type<op::v0::Constant>()});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto gather_nd_node = std::dynamic_pointer_cast<ov::op::util::GatherNDBase>(m.get_match_root());
+        if (!gather_nd_node) {
+            return false;
+        }
+        const auto original_indices = gather_nd_node->get_input_source_output(1);
+        const auto const_indices_node =
+            std::dynamic_pointer_cast<ngraph::opset8::Constant>(original_indices.get_node_shared_ptr());
+        if (!const_indices_node) {
+            return false;
+        }
+        const auto& const_indices_values = const_indices_node->cast_vector<int64_t>();
+        if (const_indices_values.size() == 0) {
+            return false;
+        }
+        const auto data = gather_nd_node->get_input_source_output(0);
+        const auto data_shape = data.get_shape();
```

Comment:
Please use `get_partial_shape` and check if static (or only `rank` if it is enough)

Reply:
Done.